### PR TITLE
metrics(dm, engine): add job_id variable to dm grafana

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/6985

### What is changed and how it works?
1. Add job_id variable to dm grafana.

#### Tests 

 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/jobmaster/dm/task_manager.go [hunk 1]
```diff
@@ -51,19 +52,27 @@ type TaskManager struct {
...
+	pLogger *zap.Logger,
+	metricFactory promutil.Factory,
+) *TaskManager {
 	taskManager := &TaskManager{
+		jobID:         jobID,
 		DefaultTicker: ticker.NewDefaultTicker(taskNormalInterval, taskErrorInterval),
 		jobStore:      jobStore,
 		logger:        pLogger.With(zap.String("component", "task_manager")),
 		messageAgent:  messageAgent,
 		gaugeVec: metricFactory.NewGaugeVec(
```
Comment (CharlesCheung96) [inline engine/jobmaster/dm/task_manager.go:69]: @lance6716 @okJiang Please review this change.

Comment (lance6716) [reply engine/jobmaster/dm/task_manager.go:69]: I think it's OK

Comment (okJiang) [reply engine/jobmaster/dm/task_manager.go:69]: ok for me

File: engine/jobmaster/dm/task_manager.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ var (
 type TaskManager struct {
 	*ticker.DefaultTicker
 
+	jobID        string
```
Comment (maxshuang) [inline engine/jobmaster/dm/task_manager.go:43]: Is this field just a compatible field and only be used in `op`?
May be we can use a `fakeJobID` or `dummyJobID`.

Comment (CharlesCheung96) [reply engine/jobmaster/dm/task_manager.go:43]: This is used to extract the job_id label from the monitor.

File: dm/metrics/grafana/DM-Monitor-Professional.json [hunk 3]
```diff
@@ -200,11 +200,11 @@
       "targets": [
         {
           "exemplar": true,
-          "expr": "dm_worker_task_state{task=~\"$task\",source_id=~\"$source\"}",
+          "expr": "dm_worker_task_state{job_id=~\"$job_id\",task=~\"$task\",source_id=~\"$source\"}",
           "format": "time_series",
           "interval": "$interval",
           "intervalFactor": 1,
-          "legendFormat": "{{task}}-{{worker}}",
+          "legendFormat": "{{task}}-{{source_id}}",
```
Comment (lance6716) [inline dm/metrics/grafana/DM-Monitor-Professional.json:207]: value of work is different with source_id, 127.0.0.1:8262 vs mysql-01

File: dm/metrics/grafana/DM-Monitor-Professional.json [hunk 4]
```diff
@@ -5617,7 +5620,7 @@
             "h": 7,
             "w": 6,
             "x": 0,
-            "y": 12
+            "y": 102
```
Comment (lance6716) [inline dm/metrics/grafana/DM-Monitor-Professional.json:5623]: is this expected?

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dm/metrics/grafana/DM-Monitor-Professional.json (hunk 3 above)
Comment (CharlesCheung96) [reply dm/metrics/grafana/DM-Monitor-Professional.json]: change lengend format to `{{task}}-{{source_id}}-{{worker}}`

**Review by maxshuang — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**